spec/lib/pending_assigns_reminder_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe PendingAssignsReminder do
+  let(:user) { Fabricate(:user) }
+
+  it 'Do not create a reminder if the user has 0 assigned topics' do
+    assert_reminder_not_created
+  end
+
+  it 'Do not create a reminder if the user only has one task' do
+    post = Fabricate(:post)
+    TopicAssigner.new(post.topic, user).assign(user)
+
+    assert_reminder_not_created
+  end
+
+  def assert_reminder_not_created
+    expect { subject.remind(user) }.to change { Post.count }.by(0)
+  end
+
+  describe 'When the user has multiple tasks' do
+    let(:system) { Discourse.system_user }
+
+    before do
+      @post = Fabricate(:post)
+      @another_post = Fabricate(:post)
+      TopicAssigner.new(@post.topic, user).assign(user)
+      TopicAssigner.new(@another_post.topic, user).assign(user)
+      @assigned_posts = 2
+    end
+
+    it 'Creates a reminder for a particular user' do
+      subject.remind(user)
+
+      created_post = Post.includes(topic: %i[topic_allowed_users]).last
+
+      assert_remind_was_created_correctly(created_post.topic, created_post)
+    end
+
+    it 'Sets the timestamp of the last reminder' do
```

Comment:
I would just combine this test with the one above since the outcome is the same.